### PR TITLE
cxxtest 4.3

### PR DESCRIPTION
**Pull request by dsdolzhenko**
Unit testing framework for C++ https://github.com/CxxTest/cxxtest

### CODE REVIEW

File: Library/Formula/cxxtest.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "356d0f4810e8eb5c344147a0cca50fc0d84122c286e7644b61cb365c2ee22083"
+
+  def install
+    libexec.install Dir["*"]
+    bin.install_symlink "#{libexec}/bin/cxxtestgen"
+    include.install_symlink "#{libexec}/cxxtest"
+  end
+
+  test do
+    system "#{bin}/cxxtestgen", "--version"
```
Comment (bfontaine) [inline Library/Formula/cxxtest.rb]: Could you add a test that does something more substantial than e.g. `--version` or `--help`? Thanks!

File: Library/Formula/cxxtest.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    libexec.install Dir["*"]
+    bin.install_symlink "#{libexec}/bin/cxxtestgen"
+    include.install_symlink "#{libexec}/cxxtest"
+  end
+
+  test do
+    testfile = testpath/"MyTestSuite1.h"
+    testfile.write <<-EOS.undent
+      #include <cxxtest/TestSuite.h>
+
```
Comment (bfontaine) [inline Library/Formula/cxxtest.rb]: Please remove the trailing spaces here.

File: Library/Formula/cxxtest.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      
+      class MyTestSuite1 : public CxxTest::TestSuite {
+      public:
+          void testAddition(void) {
+              TS_ASSERT(1 + 1 > 1);
+              TS_ASSERT_EQUALS(1 + 1, 2);
+          }
+      };
+    EOS
+
```
Comment (bfontaine) [inline Library/Formula/cxxtest.rb]: Here too.

File: Library/Formula/cxxtest.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      public:
+          void testAddition(void) {
+              TS_ASSERT(1 + 1 > 1);
+              TS_ASSERT_EQUALS(1 + 1, 2);
+          }
+      };
+    EOS
+    
+    system bin/"cxxtestgen", "--error-printer", "-o", testpath/"runner.cpp", testfile
+    system "g++", "-o", testpath/"runner", testpath/"runner.cpp"
```
Comment (bfontaine) [inline Library/Formula/cxxtest.rb]: Use `ENV.cxx` instead of `"g++"`.

**Comment by mikemcquaid** [comment]
CC @adamv who I think has talked about vendoring stuff like this before.

**Comment by dsdolzhenko** [comment]
@mikemcquaid, did you mean his comment on https://github.com/Homebrew/homebrew/pull/17740 ?
I've updated formula, now it uses python setup.py script to install binaries, as in most python application in homebrew

File: Library/Formula/cxxtest.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
+class Cxxtest < Formula
+  desc "CxxTest Unit Testing Framework"
+  homepage "http://cxxtest.com"
+  url "https://github.com/CxxTest/cxxtest/archive/4.3.tar.gz"
+  sha256 "356d0f4810e8eb5c344147a0cca50fc0d84122c286e7644b61cb365c2ee22083"
+
```
Comment (mikemcquaid) [inline Library/Formula/cxxtest.rb]: Needs a `depends_on :python if MacOS.version <= :snow_leopard`.

**Comment by dsdolzhenko** [comment]
> Needs a depends_on :python if MacOS.version <= :snow_leopard.

done

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `cxxtest 4.3 (new formula)`.

**Pull request closed without merging**